File: LeetCode/Unknown/프로그래머스/3/12938. 최고의 집합/최고의 집합.java
class Solution {
    public int[] solution(int n, int s) {
        int[] answer = new int[n];
        if (n > s) return new int[] {-1};
        
        int num = s / n;
        for (int i = 0; i < n; i++) {
            answer[i] = num;
        }
        
        int over = s - num * n;
        int index = n - 1;
        
        while (over != 0) {
            answer[index--]++;
            over--;
        }
        
        return answer;
    }
}

File: LeetCode/Unknown/프로그래머스/3/12938. 최고의 집합/최고의 집합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int N, int S) {
    if (S < N) return {-1};
    
    int num = S / N;
    vector<int> answer(N, num);
    
    int over = S - num * N;
    int index = N - 1;
    
    while (over--) {
        answer[index--]++;
    }
    
    return answer;
}